nupic/research/frameworks/dendrites/dendritic_layers.py

```diff
@@ -68,31 +67,36 @@ def __init__(
         self.rezero_weights()
 
     def rezero_weights(self):
+        """Set the previously selected weights to zero."""
         super().rezero_weights()
         if self.segments is not None:  # only none at beginning of init
             self.segments.rezero_weights()
 
     def apply_dendrites(self, y, dendrite_activations):
```

Comment:
You should add the `@abc.abstractmethod` decorator (here is an [example](https://github.com/numenta/nupic.torch/blob/0f0e2452dd648ef0421d53d872c5aaa4b6a7feae/nupic/torch/modules/sparse_weights.py#L63)). This ensures that the `apply_dendrites` function is defined by the time a `DendriticLayerBase` subclass is instantiated. Otherwise, it will only through an error much later on when `forward` is called.

Reply:
`DendriticLayerBase` needs to also be a subclass of `SparseWeights`, so it would actually be the following, correct?

```
class DendriticLayerBase(SparseWeights, metaclass=abc.ABCMeta):
```

Reply:
Ah true